From dae762865d2b344345a4ede435b5befcf6fdaad7 Mon Sep 17 00:00:00 2001
From: David Szendrei <618011+pocker@users.noreply.github.com>
Date: Fri, 2 Oct 2026 06:42:11 +0200
Subject: [PATCH] winebth.sys: Define WIN32_NO_STATUS after including
 ntstatus.h.

Proton's winnt.h still defines the STATUS_* codes unless WIN32_NO_STATUS
is set, so including both headers fails to build with -Werror.
---
 dlls/winebth.sys/dbus.c          | 1 +
 dlls/winebth.sys/unixlib.c       | 1 +
 dlls/winebth.sys/unixlib_priv.h  | 1 +
 dlls/winebth.sys/winebluetooth.c | 1 +
 dlls/winebth.sys/winebth.c       | 1 +
 5 files changed, 5 insertions(+)

diff --git a/dlls/winebth.sys/dbus.c b/dlls/winebth.sys/dbus.c
index 3cb7ea38..41bf9890 100644
--- a/dlls/winebth.sys/dbus.c
+++ b/dlls/winebth.sys/dbus.c
@@ -39,6 +39,7 @@
 #endif
 
 #include <ntstatus.h>
+#define WIN32_NO_STATUS
 #include <windef.h>
 #include <winternl.h>
 #include <winbase.h>
diff --git a/dlls/winebth.sys/unixlib.c b/dlls/winebth.sys/unixlib.c
index 3705c1b8..4de4855b 100644
--- a/dlls/winebth.sys/unixlib.c
+++ b/dlls/winebth.sys/unixlib.c
@@ -31,6 +31,7 @@
 #include <pthread.h>
 
 #include <ntstatus.h>
+#define WIN32_NO_STATUS
 #include <winternl.h>
 #include <winbase.h>
 #include <windef.h>
diff --git a/dlls/winebth.sys/unixlib_priv.h b/dlls/winebth.sys/unixlib_priv.h
index 84dc95a0..1c8fe190 100644
--- a/dlls/winebth.sys/unixlib_priv.h
+++ b/dlls/winebth.sys/unixlib_priv.h
@@ -23,6 +23,7 @@
 #define __WINE_WINEBTH_UNIXLIB_PRIV_H
 
 #include <ntstatus.h>
+#define WIN32_NO_STATUS
 #include <windef.h>
 
 #include <wine/list.h>
diff --git a/dlls/winebth.sys/winebluetooth.c b/dlls/winebth.sys/winebluetooth.c
index a9dacd4d..bc28eb55 100644
--- a/dlls/winebth.sys/winebluetooth.c
+++ b/dlls/winebth.sys/winebluetooth.c
@@ -22,6 +22,7 @@
 #include <stdarg.h>
 
 #include <ntstatus.h>
+#define WIN32_NO_STATUS
 
 #include <windef.h>
 #include <winbase.h>
diff --git a/dlls/winebth.sys/winebth.c b/dlls/winebth.sys/winebth.c
index 25f67de2..a46a3044 100644
--- a/dlls/winebth.sys/winebth.c
+++ b/dlls/winebth.sys/winebth.c
@@ -24,6 +24,7 @@
 #include <assert.h>
 
 #include <ntstatus.h>
+#define WIN32_NO_STATUS
 #include <windef.h>
 #include <winbase.h>
 #include <winternl.h>
